File: app/src/main/java/com/hjc/base/ui/other/dialog/DialogActivity.java
package com.hjc.base.ui.other.dialog;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.hjc.base.R;
import com.hjc.base.base.activity.BaseActivity;
import com.hjc.base.constant.RoutePath;
import com.hjc.base.http.helper.RxSchedulers;
import com.hjc.base.widget.bar.TitleBar;
import com.hjc.base.widget.dialog.ConfirmDialog;
import com.hjc.base.widget.dialog.LoadingDialog;

import java.util.concurrent.TimeUnit;

import butterknife.BindView;
import io.reactivex.Observable;

@Route(path = RoutePath.URL_DIALOG)
public class DialogActivity extends BaseActivity {
    @BindView(R.id.title_bar)
    TitleBar titleBar;
    @BindView(R.id.btn_loading)
    Button btnLoading;
    @BindView(R.id.btn_animation)
    Button btnAnimation;

    private LoadingDialog loadingDialog;

    @Override
    public int getLayoutId() {
        return R.layout.activity_dialog;
    }

    @Override
    public void initView() {
        loadingDialog = LoadingDialog.newInstance();
    }

    @Override
    public void initData(Bundle savedInstanceState) {

    }

    @Override
    public void addListeners() {
        btnLoading.setOnClickListener(this);
        btnAnimation.setOnClickListener(this);

        titleBar.setOnViewLeftClickListener(view -> finish());
    }

    @Override
    public void onSingleClick(View v) {
        switch (v.getId()) {
            case R.id.btn_loading:
                loadingDialog.showDialog(getSupportFragmentManager());
                Observable.timer(2, TimeUnit.SECONDS)
                        .compose(RxSchedulers.ioToMain())
                        .subscribe(aLong -> loadingDialog.dismissDialog());
                break;


            case R.id.btn_animation:
                ConfirmDialog.newInstance()
                        .setAnimStyle(R.style.dialog_anim_bottom)
                        .showDialog(getSupportFragmentManager());
                break;

            default:
                break;
        }
    }
}

File: module-other/src/main/java/com/hjc/module_other/ui/dialog/DialogActivity.kt
package com.hjc.module_other.ui.dialog

import android.app.AlertDialog
import android.content.DialogInterface
import android.os.Bundle
import android.view.View
import com.alibaba.android.arouter.facade.annotation.Autowired
import com.alibaba.android.arouter.facade.annotation.Route
import com.blankj.utilcode.util.LogUtils
import com.gyf.immersionbar.ImmersionBar
import com.hjc.library_base.activity.BaseActivity
import com.hjc.library_base.utils.ActivityHelper
import com.hjc.library_common.router.ServicePath
import com.hjc.library_common.router.path.RouteOtherPath
import com.hjc.library_common.router.services.ILoginService
import com.hjc.library_common.viewmodel.CommonViewModel
import com.hjc.library_widget.bar.OnViewLeftClickListener
import com.hjc.module_other.R
import com.hjc.module_other.databinding.OtherActivityDialogBinding
import com.hjc.module_other.dialog.TestDialog
import com.hjc.module_other.dialog.UpdateDialog

/**
 * @Author: HJC
 * @Date: 2021/7/7 22:44
 * @Description: 弹框演示
 */
@Route(path = RouteOtherPath.URL_DIALOG)
class DialogActivity : BaseActivity<OtherActivityDialogBinding, CommonViewModel>() {

    @JvmField
    @Autowired(name = ServicePath.LOGIN_SERVICE)
    var loginService: ILoginService? = null

    override fun getLayoutId(): Int {
        return R.layout.other_activity_dialog
    }

    override fun createViewModel(): CommonViewModel? {
        return null
    }

    override fun getImmersionBar(): ImmersionBar? {
        return ImmersionBar.with(this)
            .fitsSystemWindows(true)
            .statusBarColor(R.color.other_color)
    }

    override fun initData(savedInstanceState: Bundle?) {
        LogUtils.e("username: ${loginService?.getUsername()}")
    }

    override fun addListeners() {
        mBindingView.onClickListener = this

        mBindingView.titleBar.setOnViewLeftClickListener(object : OnViewLeftClickListener {
            override fun onViewLeftClick(view: View?) {
                finish()
            }
        })
    }

    override fun onSingleClick(v: View?) {
        when (v?.id) {
            R.id.btn1 -> {
                AlertDialog.Builder(this)
                    .setTitle("确认退出App吗")
                    .setNegativeButton("取消") { dialog: DialogInterface?, _: Int -> dialog?.dismiss() }
                    .setPositiveButton("确定") { dialog: DialogInterface?, _: Int ->
                        dialog?.dismiss()
                        ActivityHelper.finishAllActivities()
                    }
                    .show()
            }

            R.id.btn2 -> UpdateDialog.newInstance().showDialog(supportFragmentManager)

            R.id.btn3 -> TestDialog.newInstance().showDialog(supportFragmentManager)

            else -> {
            }
        }
    }

}